### ISSUE TITLE
please disable multi-threaded linking or port upstream patch for LLVM wasm-ld threadpool deadlock issue

### ISSUE THREAD

**Issue by godmar** [EOSIO/eosio.cdt#741]
A brief follow-up on [my note](https://github.com/EOSIO/eosio.cdt/issues/123#issuecomment-552991542) in #123  - eosio.cdt 1.6.1 runs with parallel linking enabled, but there's a deadlock in LLVM's threadpool that doesn't allow the parallel writing of sections and custom sections in the WASM backend. See [here](https://bugs.llvm.org/show_bug.cgi?id=41508) for the upstream bug that was fixed in LLVM 8.0.1.

Other projects affected by it have disabled parallel linking, e.g. `ziglang` disabled parallel linking, see [here for their bugfix](https://github.com/ziglang/zig/commit/6692cbbe18ad108affccfb9cb85d53e2583408ec). You can do the same, or possibly backport [their fix](https://github.com/llvm/llvm-project/commit/5081e41bdae2eb14a3f3eb8810263f9fea8fc7c1) - which is just a few lines of code.

I've copied and pasted more details, including a backtrace of when eosio-cpp deadlocks, [here](https://gist.github.com/godmar/a52562a04acc8a3e529fdf4da08ef9e6)

**Comment by larryk85** [comment]
This is on the radar, we will be moving to LLVM 9 as the backbone next release.  This will remove this issue, and hopefully not create new ones.

**Comment by godmar** [comment]
This is a really unfortunate issue. I'm right now hitting this reproducibly on a t2.medium EC2 instance (which does have 2 virtual CPUs).  Of course, the deadlock occurs if these cores aren't scheduled often enough in the underlying multi-tenant environment, which makes the entire enterprise unpredictable and pretty annoying.

**Comment by godmar** [comment]
Here's a work-around that doesn't require patching and rebuilding the cdt:
```
cd /usr/opt/eosio.cdt/1.6.1/bin
sudo mv wasm-ld wasm-ld.real
```
then create a file `/usr/opt/eosio.cdt/1.6.1/bin/wasm-ld` like so:
```
#!/bin/bash

echo "Running " `dirname $0`/wasm-ld.real --no-threads "$@"
`dirname $0`/wasm-ld.real --no-threads "$@"
```

Then `sudo chmod +x /usr/opt/eosio.cdt/1.6.1/bin/wasm-ld` and then use `eosio-cpp` as before.

**Comment by jeffreyssmith2nd** [comment]
This should be resolved in develop with #896
